elastic/kibana
# [Reporting] use Task Manager for retrying failed job execution attempts

In this PR I've removed the `reports:monitor` task from reporting. That task was responsible for checking the number of attempts to execute a report and reschedule or fail the report accordingly.

# How to test

In order to test the rescheduling of the report I added the following in `kibana.yml`

```
xpack.reporting.capture.maxAttempts: 2
```

And then I modified the `packages/kbn-reporting/export_types/csv/csv_searchsource.ts` file to throw on certain attempts.

```diff
@@ -40,6 +40,8 @@ interface CsvSearchSourceExportTypeStartDeps extends BaseExportTypeStartDeps {
   discover: DiscoverServerPluginStart;
 }

+let requestCounter = 0;
+
 export class CsvSearchSourceExportType extends ExportType<
   JobParamsCSV,
   TaskPayloadCSV,
@@ -75,6 +77,13 @@ export class CsvSearchSourceExportType extends ExportType<
     cancellationToken: CancellationToken,
     stream: Writable
   ) => {
+    console.log(`Running CSV task, attempt ${fakeRequestCounter}`);
+    if (requestCounter < 3) {
+      requestCounter += 1;
+      await new Promise((resolve) => setTimeout(resolve, 5000));
+      throw new Error('Fake error');
+    }
+
```

* We can manually kill the Kibana server during a run, it should re-schedule a task on the next server start. **Note!** it can take up to 10 minutes for the task to be rescheduled after Kibana starts
* We can change `requestCounter < 1` to simulate a first fail and then the second that succeeds and generates the CSV report 

Fixes https://github.com/elastic/kibana/issues/170462
